utils/model_wrappers/langchain_chat_models.py

```diff
@@ -1008,26 +1012,77 @@ class ChatSambaStudio(BaseChatModel):
     Stream:
         .. code-block:: python
 
-        for chunk in chat.stream(messages):
-            print(chunk.content, end="", flush=True)
+            for chunk in chat.stream(messages):
+                print(chunk.content, end="", flush=True)
 
     Async:
         .. code-block:: python
 
-        response = chat.ainvoke(messages)
-        await response
+            response = chat.ainvoke(messages)
+            await response
+
+    Tool calling:
+        .. code-block:: python
+
+            from pydantic import BaseModel, Field
+
+            class GetWeather(BaseModel):
+                '''Get the current weather in a given location'''
+
+                location: str = Field(
+                    ...,
+                    description="The city and state, e.g. Los Angeles, CA"
+                )
```

Comment:
Suggestion to include a complete implementation of a tool class (e.g., GetWeather) or link to a complete implementation (e.g., in function calling AISK)